frontend/src/layout/navigation/TopBar/notificationsLogic.tsx

```diff
@@ -89,28 +110,82 @@ export const notificationsLogic = kea<notificationsLogicType>([
             if (!values.isNotificationPopoverOpen) {
                 clearTimeout(values.markReadTimeout)
             } else {
-                if (values.importantChanges?.[0]) {
+                if (values.notifications?.[0]) {
+                    const bookmarkDate = values.notifications.reduce((a, b) =>
+                        a.created_at.isAfter(b.created_at) ? a : b
+                    ).created_at
                     actions.setMarkReadTimeout(
                         window.setTimeout(async () => {
-                            const bookmarkDate = values.importantChanges[0].created_at.toISOString()
                             await api.create(
                                 `api/projects/${teamLogic.values.currentTeamId}/activity_log/bookmark_activity_notification`,
                                 {
-                                    bookmark: bookmarkDate,
+                                    bookmark: bookmarkDate.toISOString(),
                                 }
                             )
-                            actions.markAllAsRead()
+                            actions.markAllAsRead(bookmarkDate.toISOString())
                         }, MARK_READ_TIMEOUT)
                     )
                 }
             }
         },
     })),
     selectors({
-        unread: [(s) => [s.importantChanges], (importantChanges) => importantChanges.filter((ic) => ic.unread)],
+        notifications: [
+            (s) => [s.importantChanges],
+            (importantChanges): HumanizedActivityLogItem[] => {
+                const importantChangesHumanized = humanize(importantChanges?.results || [], describerFor, true)
+
+                let changelogNotification: ChangelogFlagPayload | null = null
+                const flagPayload = posthog.getFeatureFlagPayload('changelog-notification')
+                if (!!flagPayload) {
+                    changelogNotification = {
+                        markdown: flagPayload['markdown'],
+                        notificationDate: dayjs(flagPayload['notificationDate']),
+                    } as ChangelogFlagPayload
```

Comment:
Is this where the new `satisfies` operator can come in handy 🤔

(I never used it myself...)

Reply:
Ooh, did we merge that upgrade....

Been meaning to wrap my head around it too https://twitter.com/mattpocockuk/status/1631249088795533315

Reply:
`satisfies` would behave identically to `as` here I believe, since we don't know what the values are gonna be at runtime anyway. But `satisfies` is great for values where we know the value at compile-time AND want to be sure that value always conforms to an interface.